sunpy/net/helio/tests/test_helio.py

```diff
@@ -259,16 +260,39 @@ def test_link_test_on_urlerror(mock_link_test):
     link_test('') is None
 
 
+# Check if a server is active or not using the isOpen functionality
+def isOpen(ip, port, timeout):
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.settimeout(timeout)
+    try:
+        s.connect((ip, int(port)))
+        s.shutdown(socket.SHUT_RDWR)
+        return True
+    except socket.gaierror:
+        return False
+    finally:
+        s.close()
+
+
 @pytest.mark.remote_data
 @pytest.fixture(scope="session")
 def client():
     working_links = list(filter(link_test, webservice_parser()))
```

Comment:
In theory this link_test should already filter out any bad URLS. I wonder if we need to check that works and then if this returns nothing we pytest.xfail?

Reply:
Maybe it passes the urls whose servers are down at the moment, if I am not wrong i guess that was the issue that server when down thus resulted in failure. isOpen ensures that not be the issue.

Reply:
Then we could fix `link_test` instead of creating a new function?

Reply:
so shall i implement this function in link_test if you want. or if we wish to keep this   way also, i mean link_test will test the link while isOpen will check if the server is up. I will update it accordingly if you recommend.

Reply:
I don't understand why we need another function to do the test? Why can't link_test do it all?

Reply:
okay i got your point... test_link itself can do check this functionality thanks for pointing me out i will remove this redundant function then